frontend/packages/dev-console/src/components/monitoring/alerts/SilenceDurationDropdown.tsx

```diff
@@ -48,18 +61,43 @@ const SilenceDurationDropDown: React.FC<SilenceDurationDropDownProps> = ({ rule
         matchers,
         comment: '',
       };
-
-      coFetchJSON.post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload);
+      setSilencing(true);
+      coFetchJSON
+        .post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload)
+        .then(() => {
+          // eslint-disable-next-line promise/no-nesting
+          coFetchJSON(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`)
+            .then((silences) => {
+              clonedRule.silencedBy = _.filter(
+                silences,
+                (s) =>
+                  s.status.state === SilenceStates.Active && _.some(clonedRule.alerts, isSilenced),
+              );
+              if (!_.isEmpty(clonedRule.silencedBy)) {
+                _.each(clonedRule.alerts, (a) => (a.state = AlertStates.Silenced));
+                clonedRule.state = RuleStates.Silenced;
+              }
+              const updatedRules = [...rules, ...[clonedRule]];
+              dispatch(monitoringSetRules('devRules', updatedRules, 'dev'));
+              setSilencing(false);
```

Comment:
` setSilencing(false);  
`
This needs to be set on the catch block as well, or else loader is shown there indefinitely even after the error modal is closed.
eg: 
![image](https://user-images.githubusercontent.com/9964343/94775688-f18ae700-03dd-11eb-9889-3c27c716bb40.png)